File: Java/Treinamento GFT/Aula 08 - Prova 2/src/br/com/gft/model/Funcionario.java
package br.com.gft.model;

public abstract class Funcionario {
	
	public Funcionario(String nome, int idade, double salario) {
		this.nome = nome;
		this.idade = idade;
		this.salario = salario;
	}
	
	String nome;
	int idade;
	double salario;
	
	public double getSalario() {
		return salario;
	}
	
	public abstract double bonificacao();
	
	@Override
	public String toString() {
		return "---------------------\nNome: " + this.nome + "\nIdade: " + this.idade + "\nSalario: " + this.salario + "\nBonificação: " + bonificacao();
	}
}

File: C#/Treinamento GFT/Aula 08 - Prova Funcionarios/Aula 08 - Prova Funcionarios/Funcionario .cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Aula_08___Prova_Funcionarios
{
    abstract class Funcionario
    {
        public Funcionario(string nome, int idade, double salario)
        {
            Nome = nome;
            Idade = idade;
            Salario = salario;
        }
        public string Nome { get; set; }
        public int Idade { get; set; }
        public double Salario { get; set; }

        public virtual double Bonificacao()
        {
            return Salario;
        }

        public override string ToString()
        {
            return "-----------------\nNome: " + Nome + "\nIdade: " + Idade + "\nSalario: R$" + Salario + "\nBonificação: R$" + Bonificacao();
        }

    }
}
